### PR TITLE
Request for Ansible AWX to charts.

### PR DESCRIPTION
**Pull request by Najeongho**
What is AWX?
- AWX provides a web-based user interface, REST API, and task engine built on top of Ansible.

1. What this PR does 
- Provides a web-based Ansible GUI for easy configuration.

2. why we need it
- Configure AWX automatically at once.
- Apply automatically inventory gathering function (Ex. IBM Cloud)
- Provides automated deployment through Linux SSH.
- Provides automated deployment through Windows Certificate.

### CODE REVIEW

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please have a look at our review guidelines and follow best practices. Also, please compare with what you get from `helm create`using Helm 2.8+ and update accordingly.

https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

**Comment by Najeongho** [comment]
@unguiculus 

I modified everything as you guided.
And I performed the test in version 2.9.1 of Helm.
Of course, I have confirmed that it works well.

I want you to make sure this chart works.
Please make sure that it meets the conditions to be merged to master.

File: incubator/awx/templates/db_deployment.yaml [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
     role: {{ .Values.dbRole }}
 spec:
-  replicas: {{ .Values.dbReplicas }}
+  replicas: {{ .Values.dbReplicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "awx.name" . }}
+      release: {{ .Release.Name }}
```
Comment (unguiculus) [inline incubator/awx/templates/db_deployment.yaml:16]: All you deployments have the same selectors. You need to add the `role` label. You may want to rename `role` to `component` if possible to be consistent with a common practice here.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/awx/templates/db_deployment.yaml (hunk 1 above)
Comment (Najeongho) [reply incubator/awx/templates/db_deployment.yaml:16]: @unguiculus 

I changed everything as you said.

Please check the changes.

File: incubator/awx/templates/awx-task_deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    chart: {{ template "awx.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    component: {{ .Values.taskComponent }}
+spec:
+  replicas: {{ .Values.taskReplicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "awx.name" . }}
+      release: {{ .Release.Name }}
```
Comment (unguiculus) [inline incubator/awx/templates/awx-task_deployment.yaml:16]: Add `component` label.

File: incubator/awx/templates/cache_deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    chart: {{ template "awx.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    component: {{ .Values.cacheComponent }}
+spec:
+  replicas: {{ .Values.cacheReplicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "awx.name" . }}
+      release: {{ .Release.Name }}
```
Comment (unguiculus) [inline incubator/awx/templates/cache_deployment.yaml:16]: Add `component` label.

File: incubator/awx/templates/db_deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    chart: {{ template "awx.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    component: {{ .Values.dbComponent }}
+spec:
+  replicas: {{ .Values.dbReplicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "awx.name" . }}
+      release: {{ .Release.Name }}
```
Comment (unguiculus) [inline incubator/awx/templates/db_deployment.yaml:16]: Add `component` label.

File: incubator/awx/templates/queue_deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    chart: {{ template "awx.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    component: {{ .Values.queueComponent }}
+spec:
+  replicas: {{ .Values.queueReplicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "awx.name" . }}
+      release: {{ .Release.Name }}
```
Comment (unguiculus) [inline incubator/awx/templates/queue_deployment.yaml:16]: Add `component` label.

File: incubator/awx/templates/web_deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    chart: {{ template "awx.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    component: {{ .Values.webComponent }}
+spec:
+  replicas: {{ .Values.webReplicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "awx.name" . }}
+      release: {{ .Release.Name }}
```
Comment (unguiculus) [inline incubator/awx/templates/web_deployment.yaml:16]: Add `component` label.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by Najeongho** [comment]
@unguiculus

I changed everything as you said. (Add component label.)


Please check the changes.

File: incubator/awx/templates/web_deployment.yaml (hunk 6 above)
Comment (Najeongho) [reply incubator/awx/templates/web_deployment.yaml:16]: @unguiculus

I changed everything as you said. (Add component label.)

I pushed my sources.

Please check the changes.

**Review by unguiculus — CHANGES_REQUESTED** [review]
One last thing and sorry for the delay. Please make list indentation consistent. Either use zero or two spaces throughout the chart.

**Review by unguiculus — APPROVED** [review]

File: incubator/awx/templates/cache_deployment.yaml [hunk 7]
```diff
@@ -30,16 +30,16 @@ spec:
         - name: {{ .Values.cacheName }}
           containerPort: {{ .Values.cacheService.port }}
         resources:
-{{ toYaml .Values.cacheResources | indent 10 }}
-    {{- with .Values.cacheNodeSelector }}
+          {{ toYaml .Values.cacheResources | indent 10 }}
```
Comment (unguiculus) [inline incubator/awx/templates/cache_deployment.yaml]: You should not indent `toYaml` statements. This won't work. Please revert these changes throughout the chart.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/awx/templates/cache_deployment.yaml (hunk 7 above)
Comment (Najeongho) [reply incubator/awx/templates/cache_deployment.yaml]: I changed everything as you said.

Please review it again.

File: incubator/awx/values.yaml [hunk 8]
```diff
@@ -158,7 +158,7 @@ cacheReplicaCount: 1
 ## Cache image repository and version
 cacheImage:
   repository: memcached
-  tag: alpine
+  tag: latest
```
Comment (unguiculus) [inline incubator/awx/values.yaml]: Please use a stable tag

**Review by unguiculus — CHANGES_REQUESTED** [review]
Again, please make list indentation consistent. Either use zero or two spaces throughout the chart.

**Comment by joaocc** [comment]
Any updates on this? Can this be merged already? Thx
